jb/testing/create_triangle_timeseries.hpp

```diff
@@ -9,15 +14,44 @@ namespace testing {
  */
 template <typename timeseries>
 void create_triangle_timeseries(int nsamples, timeseries& ts) {
-  typedef typename timeseries::value_type value_type;
-  ts.resize(nsamples);
+  resize_if_applicable(ts, nsamples);
+  using value_type = typename timeseries::value_type;
   float p4 = nsamples / 4;
   for (int i = 0; i != nsamples / 2; ++i) {
     ts[i] = value_type((i - p4) / p4);
     ts[i + nsamples / 2] = value_type((p4 - i) / p4);
   }
 }
 
+/**
+ * Create timeseries where the values look like a triangle
+ * @param nsamples samples on each timeseries
+ * @param ts multi array containing timeseries to be filled as triangle
+ *
+ * @tparam T type of the timeseries field
+ * @tparam K timeseries dimensionality
+ */
+
+template <typename T, std::size_t K>
+void create_triangle_timeseries(
+    int nsamples, jb::fftw::aligned_multi_array<T, K>& ts) {
```

Comment:
You can make this work for non-aligned arrays too... please consider making that change.